app/services/physical_holdings_markup_builder.rb

```diff
@@ -293,21 +293,31 @@ def self.location_services_block(adapter, holding_id, location_rules, link, hold
 
   # Generate a request label based upon the holding location
   # @param location_rules [Hash] the location for the holding
+  # @param adapter [HoldingRequestsAdapter] adapter for the SolrDocument and Bibdata API
   # @return [String] the label
-  def self.request_label(location_rules)
+  def self.request_label(location_rules, adapter)
     open_libraries = ['firestone', 'annexa', 'recap', 'marquand', 'mendel', 'stokes', 'eastasian', 'architecture', 'engineering', 'lewis']
     if aeon_location?(location_rules)
       'Reading Room Request'
     elsif open_libraries.include?(location_rules['library']['code']) && location_rules["circulates"] == true
-      'Request Pick-up or Digitization'
-    elsif open_libraries.include?(location_rules['library']['code']) &&
-          location_rules["code"] != "scsbcul" # TODO: remove this when columbia is circulating again
-      'Digitization Request'
+      if etas_restricted?(location_rules, adapter)
+        'Digitization Request'
+      else
+        'Request Pick-up or Digitization'
+      end
     else
       'Request'
     end
   end
 
+  def self.etas_restricted?(location_rules, adapter)
+    return false unless location_rules["code"] == "scsbcul"
+    bibid = adapter.document.fetch("other_id_s")
+    access_records = adapter.hathi_access(bibid)
+    return true if access_records.select { |r| r["origin"] == "CUL" }.select { |r| r["status"] == "DENY" }.present?
```

Comment:
@hackmastera I see we used slightly different logic.  https://github.com/pulibrary/requests/blob/master/app/models/requests/request.rb#L422

I wonder if we should both switch to something more like yours 
```
access_records.select { |r| r["origin"] == "CUL" && r["status"] == "DENY"  }.present?
```

There is no need to return true of false since the check would return the correct value.  Are you doing it that was to make it more explicit?

Reply:
Yep I like this suggestion. The explicit returns were just a holdover from my tdd process.